### DISCUSSION TITLE
v2.45.2.vfs.0.2

### DISCUSSION THREAD

**Discussion by dscho** [microsoft/git#668] [Announcements]
## What's Changed
* Fix [a `git restore` bug where it would fail with "did not match any file(s) known to git"](https://github.com/microsoft/VFSForGit/issues/1812) by @derrickstolee in https://github.com/microsoft/git/pull/658
* Avoid [putting GVFS packs into the local object directory of Scalar clones instead of shared object directory](https://github.com/microsoft/git/issues/645) by @derrickstolee in https://github.com/microsoft/git/pull/660
* README: adjust the currently-broken GVFS Protocol link by @dscho in https://github.com/microsoft/git/pull/661
* `git status`: print hydration level by @jeffhostetler in https://github.com/microsoft/git/pull/659
* `microsoft/git`'s release builds now use the Azure Federated login mechanism by @mjcheetham in https://github.com/microsoft/git/pull/662
* trace2: help diagnosing problems when telemetry misses events by @jeffhostetler in https://github.com/microsoft/git/pull/663
* Avoid a costly fall-back when GVFS cache servers experience network problems by @derrickstolee and @jeffhostetler in https://github.com/microsoft/git/pull/665
* Add the experimental `git survey` command to analyze (large) local repositories by @jeffhostetler in https://github.com/microsoft/git/pull/667

**Full Changelog**: https://github.com/microsoft/git/compare/v2.45.2.vfs.0.1...v2.45.2.vfs.0.2

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/git/releases/tag/v2.45.2.vfs.0.2'>v2.45.2.vfs.0.2</a>.</em>

**Comment by dscho** [comment]
For the record: I just promoted this to a full release and marked it as the latest version.

I heard of no problems other than a minor issue with the new, experimental `git survey` command where it may print unexpected blob sizes. Since this is experimental and subject to be iterated on rapidly, I did not consider this a blocker.

**Reply by derrickstolee** [reply]
FYI: I just pulled this release back to a pre-release due to a performance regression in VFS for Git enabled repositories. It seems that `git restore .` will overhydrate the repository due to the change in #658.

**Reply by dscho** [reply]
The [bug fix PR](https://github.com/microsoft/git/pull/674) was just merged. I think we'll want a v2.45.2.vfs.0.3 with the fix, right?

**Reply by dscho** [reply]
v2.45.2.vfs.0.3 is now [available as a pre-release](https://github.com/microsoft/git/releases/tag/v2.45.2.vfs.0.3).

**Reply by YehezkelShB** [reply]
For some reason, GitHub still shows v2.45.2.vfs.0.2 as pre-release instead of latest (and v2.45.2.vfs.0.1 is still linked from the main page as latest)

**Reply by dscho** [reply]
Yes, it still is a pre-release because I have not heard of anybody who tested this successfully. I want to avoid marking it as the latest release unless I have test reports to increase the confidence that this version won't break users' workflows.

For example, v2.45.2.vfs.0.2 introduced a performance regression [that forced us to pull this version](https://github.com/microsoft/git/discussions/668#discussioncomment-10034534). Despite the care we try to exercise, Git is too complex to fully prevent issues like this. However, promoting to the latest release only to then pull it again is an unideal way to deal with issues like this. It is much better that stakeholders test the pre-releases, then report back their findings, and once we gained enough confidence that it will be a solid release, we promote it to the latest release.

Please take this as very strong motivation to thoroughly test the pre-release!

**Reply by dscho** [reply]
@YehezkelShB oh, I just realized that you were talking about v2.45.2.vfs.0.2! See [this previous comment](https://github.com/microsoft/git/discussions/668#discussioncomment-10034534) in this here discussion that explains why it had to be pulled.

**Reply by YehezkelShB** [reply]
Oh, missed this. I just knew I got this version locally after auto-update :D
Thanks!
